libs/dataset-explorer/src/lib/buildScatterTemplate.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IGenericChartProps, JointDataset } from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+
+export function buildScatterTemplate(
+  jointData: JointDataset,
+  chartProps: IGenericChartProps,
+  x: any,
+  y: any,
+  customData: any
+): string {
+  let hovertemplate = "";
+  const xName = jointData.metaDict[chartProps.xAxis.property].label;
+  const yName = jointData.metaDict[chartProps.yAxis.property].label;
+  if (chartProps.xAxis) {
+    if (chartProps.xAxis.options.dither) {
+      hovertemplate += `${xName}: ${customData.X}<br>`;
+    } else {
+      hovertemplate += `${xName}: ${x}<br>`;
```

Comment:
maybe extract this into a small one line function and re-use it to output the string in the ```key: value<br>``` format?

Reply:
This is different from plotly, especially for the customized data, will figure out if there is a workaround.